features/FEATURE_BLE/targets/TARGET_CORDIO/source/CordioGattServer.cpp

```diff
@@ -510,7 +510,7 @@ ble_error_t GattServer::insert_descriptor(
 #endif // BLE_FEATURE_SECURITY
         }
 
-        if (properties & READ_PROPERTY && !(attribute_it->settings & ATTS_SET_CCC)) {
+        if (!(attribute_it->settings & ATTS_SET_CCC)) {
```

Comment:
the real condition to check is descriptor->isReadAllowed() which is already checked above